### PR TITLE
fix(results): refresh study outputs when job completed and add back button

### LINKED ISSUE #1541: [ANT-758] No data available on view results
## Description
For the areas, I only asked for the annual results in summary; in AntaresWeb impossible to see them.
But when the results is downloading on disk they are there.

## Screenshots
image

## Environment

- [x] Production
- [x] Staging

## Additional Information
Study: e1d5f541-c049-45ed-a093-b67a4f670ff3 (in Production)
Results: 20230530-0716eco

### PR DESCRIPTION
**Pull request by skamril**

### CODE REVIEW

File: webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx [hunk 1]
```diff
@@ -152,6 +154,16 @@ function ResultDetails() {
...
         left={
           <PropertiesView
+            topContent={
+              <Box
+                sx={{
+                  width: 1,
+                  px: 1,
+                }}
+              >
+                <ButtonBack onClick={() => navigate("..")} />
```
Comment (hdinia) [inline webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:164]: `navigate(-1)`

File: webapp/src/redux/ducks/studySyntheses.ts [hunk 2]
```diff
@@ -139,10 +140,12 @@ export const setStudySynthesis = createAsyncThunk<
...
 
 export const refreshStudySynthesis =
-  (event: WSMessage<GenericInfo>): AppThunk =>
+  (event: WSMessage<GenericInfo | LaunchJobDTO>): AppThunk =>
   (dispatch, getState) => {
     const state = getState();
-    const { id } = event.payload;
+    const id =
+      "study_id" in event.payload ? event.payload.study_id : event.payload.id;
+
```
Comment (hdinia) [inline webapp/src/redux/ducks/studySyntheses.ts:148]: optional: 
```javascript
const id = event.payload?.study_id ?? event.payload?.id;
```

**Review by hdinia — CHANGES_REQUESTED** [review]

File: webapp/src/redux/ducks/studySyntheses.ts (hunk 2 above)
Comment (skamril) [reply webapp/src/redux/ducks/studySyntheses.ts:148]: TS doesn't like it

File: webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx (hunk 1 above)
Comment (skamril) [reply webapp/src/components/App/Singlestudy/explore/Results/ResultDetails/index.tsx:164]: Not the same purpose

**Review by hdinia — APPROVED** [review]

**Pull request merged**